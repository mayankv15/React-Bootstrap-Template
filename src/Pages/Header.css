
.navbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: #0073b1;
    color: #fff;
    padding: 10px;
  }
  
  .nav-details {
    display: flex;
    align-items: center;
  }
  
  .nav-details a {
    color: #fff;
    text-decoration: none;
    margin-right: 20px;
    font-size: 18px;
  }
  
  .nav-details a:hover {
    color: #e0e0e0;
  }
  
  .search-container {
    display: flex;
    align-items: center;
    margin-right: 20px;
  }
  
  .search-container input[type="text"] {
    padding: 8px;
    border: none;
    border-radius: 4px;
  }
  
  .search-container button {
    background-color: #005a8e;
    color: #fff;
    border: none;
    padding: 8px 16px;
    border-radius: 4px;
    cursor: pointer;
  }
  
  .navbar-profile {
    position: relative;
  }
  
  .navbar-profile h3 {
    display: flex;
    align-items: center;
    margin: 0;
    cursor: pointer;
  }
  
  .navbar-profile button {
    background-color: transparent;
    border: none;
    color: #fff;
    font-size: 20px;
    cursor: pointer;
  }
  
  .dropdown {
    position: absolute;
    top: 100%;
    right: 0;
    background-color: #fff;
    border: 1px solid #ddd;
    border-radius: 4px;
    box-shadow: 0 4px 8px rgba(1, 0, 0, 0.1);
    padding: 10px;
    display: flex;
    flex-direction: column;
    z-index: 1;
  }
  
  .dropdown ul {
    list-style: none;
    padding: 0;
  }
  
  .dropdown a {
    text-decoration: none;
    color: #0073b1;
    font-size: 16px;
    padding: 8px;
    border-radius: 4px;
    display: block;
  }
  
  .dropdown a:hover {
    background-color: #f0f0f0;
  }
  
  