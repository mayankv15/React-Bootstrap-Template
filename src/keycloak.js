//NOTE: if i convert the keycloak.js to keycloak.ts it throws error saying Only a void function can be called with the 'new' keyword.ts(2350)

import Keycloak from 'keycloak-js';

const keycloak = new Keycloak({
    realm: `${process.env.REACT_APP_KEYCLOAK_REALM}`,
    url: `${process.env.REACT_APP_KEYCLOAK_URL}auth/`,
    clientId: `${process.env.REACT_APP_KEYCLOAK_CLIENT_ID}`,
});
export default keycloak;
