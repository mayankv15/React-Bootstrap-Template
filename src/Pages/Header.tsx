import React, { useState } from "react";
import "./Header.css";
import { useNavigate } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import { RootState } from "../redux/reducer"; // Import your RootState type

function Header() {
  const Navigate = useNavigate();
  const isLoggedIn = useSelector((state: RootState) => state.user.login); // Update RootState type
  const dispatch = useDispatch();

  const [isDropdownOpen, setIsDropdownOpen] = useState(false);

  const handleDropdownClick = () => {
    setIsDropdownOpen(!isDropdownOpen);
  };

  const handleLogoutClick = () => {
    dispatch({ type: "LOGOUT" }); // Dispatch the logout action here
    Navigate("/");
  };

  const handleProfileClick = () => {
    Navigate("/profile");
  };

  return (
    <div className="navbar">
      <link
        rel="stylesheet"
        href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
      ></link>
      <div className="nav-details">
        <div className="search-container">
          <form onSubmit={() => {}}>
            <input
              type="text"
              placeholder="Search.."
              name="search"
              onKeyUp={() => {}}
            />
            <button type="submit">search</button>
          </form>
        </div>
        <a href="/Dashboard">
          <i className="fa fa-linkedin"></i>
        </a>
        <a href="/Dashboard">
          <i className="fa fa-home"></i>
        </a>
        <a href="/My-Network">My Network</a>
        <a href="/Message">Message</a>
        <a href="/Notiifcations">Notiifcations</a>
      </div>
      <div className="navbar-profile">
        <h3>
          <button onClick={handleDropdownClick}>
            <i className="fa fa-user" aria-hidden="true"></i>
          </button>
        </h3>

        {isDropdownOpen && (
          <div className="dropdown">
            <ul>
              <button>
                <a
                  href="/"
                  onClick={handleLogoutClick}
                  style={{ color: "black" }}
                >
                  Logout
                </a>
                <a
                  href="/Profile"
                  onClick={handleProfileClick}
                  style={{ color: "black" }}
                >
                  Profile
                </a>
              </button>
            </ul>
          </div>
        )}
      </div>
    </div>
  );
}

export default Header;
