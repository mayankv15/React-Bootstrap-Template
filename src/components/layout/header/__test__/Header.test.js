import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from 'react-router-dom';
import { ReactKeycloakProvider } from "@react-keycloak/web";
import keycloak from "../../../../keycloak";
import Header from "../index"

describe("Renders Header", () => {
    it("should render Header correctly", () => {
        const component = render(
            <ReactKeycloakProvider
            initOptions={{
                onLoad: "login-required",
                checkLoginIframe: false,
            }}
            authClient={keycloak}
        >
        <Router><Header /></Router></ReactKeycloakProvider>);
        expect(component).toMatchSnapshot();
    });
});
