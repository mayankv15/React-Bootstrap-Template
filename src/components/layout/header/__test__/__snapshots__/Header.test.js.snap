// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Header should render Header correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fixed-header notify-msg"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary header notify-active MuiPaper-elevation4"
        >
          <div
            class="main-container-wrapper"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular toolbar MuiToolbar-gutters"
            >
              <h1
                class="MuiTypography-root logo MuiTypography-h6"
                style="cursor: pointer;"
              >
                <img
                  alt="brand logo"
                  src="augment-logo.svg"
                  width="50px"
                />
              </h1>
              <div
                class="menuWrapper"
              >
                <div
                  class="menusList"
                >
                  <div
                    class="menuItemsList"
                  >
                    <div
                      class="header-msg"
                    >
                      <h5
                        class="MuiTypography-root MuiTypography-h5"
                      >
                        welcome_msg
                      </h5>
                    </div>
                  </div>
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    />
                    <span
                      style="align-items: center;"
                    >
                      <button
                        aria-haspopup="true"
                        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-userProfile-1"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <div
                            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiAvatar-fallback"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                              />
                            </svg>
                          </div>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fixed-header notify-msg"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary header notify-active MuiPaper-elevation4"
      >
        <div
          class="main-container-wrapper"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular toolbar MuiToolbar-gutters"
          >
            <h1
              class="MuiTypography-root logo MuiTypography-h6"
              style="cursor: pointer;"
            >
              <img
                alt="brand logo"
                src="augment-logo.svg"
                width="50px"
              />
            </h1>
            <div
              class="menuWrapper"
            >
              <div
                class="menusList"
              >
                <div
                  class="menuItemsList"
                >
                  <div
                    class="header-msg"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5"
                    >
                      welcome_msg
                    </h5>
                  </div>
                </div>
                <div
                  style="display: flex; align-items: center;"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  />
                  <span
                    style="align-items: center;"
                  >
                    <button
                      aria-haspopup="true"
                      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-userProfile-1"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <div
                          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiAvatar-fallback"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                            />
                          </svg>
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
